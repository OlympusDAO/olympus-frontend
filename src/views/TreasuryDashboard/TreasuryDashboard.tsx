import { memo } from "react";
import "./treasury-dashboard.scss";
import { QueryClient, QueryClientProvider } from "react-query";
import { Paper, Grid, Box, Zoom, Container, useMediaQuery } from "@material-ui/core";
import { MarketCap, OHMPrice, WSOHMPrice, CircSupply, BackingPerOHM, CurrentIndex } from "./components/Metric/Metric";

import {
  TotalValueDepositedGraph,
  MarketValueGraph,
  RiskFreeValueGraph,
  ProtocolOwnedLiquidityGraph,
  OHMStakedGraph,
  RunwayAvailableGraph,
} from "./components/Graph/Graph";

const TreasuryDashboard = memo(() => {
  const isSmallScreen = useMediaQuery("(max-width: 650px)");
  const isVerySmallScreen = useMediaQuery("(max-width: 379px)");

  return (
    <div id="treasury-dashboard-view" className={`${isSmallScreen && "smaller"} ${isVerySmallScreen && "very-small"}`}>
      <Container
        style={{
          paddingLeft: isSmallScreen || isVerySmallScreen ? "0" : "3.3rem",
          paddingRight: isSmallScreen || isVerySmallScreen ? "0" : "3.3rem",
        }}
      >
        <Box className="hero-metrics">
          <Paper className="ohm-card">
            <Box display="flex" flexWrap="wrap" justifyContent="space-between" alignItems="center">
              <MarketCap />
              <OHMPrice />
              <WSOHMPrice />
              <CircSupply />
              <BackingPerOHM />
              <CurrentIndex />
            </Box>
          </Paper>
        </Box>

        <Zoom in={true}>
          <Grid container spacing={2} className="data-grid">
            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <TotalValueDepositedGraph />
              </Paper>
            </Grid>

            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <MarketValueGraph />
              </Paper>
            </Grid>

            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <RiskFreeValueGraph />
              </Paper>
            </Grid>

            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <ProtocolOwnedLiquidityGraph />
              </Paper>
            </Grid>

            {/*  Temporarily removed until correct data is in the graph */}
            {/* <Grid item lg={6} md={12} sm={12} xs={12}>
              <Paper className="ohm-card">
                <Chart
                  type="bar"
                  data={data}
                  dataKey={["holders"]}
                  headerText="Holders"
                  stroke={[theme.palette.text.secondary]}
                  headerSubText={`${data.length > 0 && data[0].holders}`}
                  bulletpointColors={bulletpoints.holder}
                  itemNames={tooltipItems.holder}
                  itemType={undefined}
                  infoTooltipMessage={tooltipInfoMessages.holder}
                  expandedGraphStrokeColor={theme.palette.graphStrokeColor}
                  scale={undefined}
                  color={undefined}
                  stroke={undefined}
                  dataFormat={undefined}
                  isPOL={undefined}
                  isStaked={undefined}
                />
              </Paper>
            </Grid> */}

            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <OHMStakedGraph />
              </Paper>
            </Grid>

            <Grid item lg={6} md={6} sm={12} xs={12}>
              <Paper className="ohm-card ohm-chart-card">
                <RunwayAvailableGraph />
              </Paper>
            </Grid>
          </Grid>
        </Zoom>
      </Container>
    </div>
  );
});

const queryClient = new QueryClient();

// Normally this would be done
// much higher up in our App.
export default () => (
  <QueryClientProvider client={queryClient}>
    <TreasuryDashboard />
  </QueryClientProvider>
);
