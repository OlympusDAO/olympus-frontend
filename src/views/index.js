export { default as Bond } from "./Bond/Bond";
export { default as ChooseBond } from "./ChooseBond/ChooseBond";
export { default as Stake } from "./Stake/Stake";
export { default as Wrap } from "./Wrap/Wrap";
export { default as Dashboard } from "./Dashboard/Dashboard";
export { default as PoolTogether } from "./33Together/33Together.jsx";
export { default as TreasuryDashboard } from "./TreasuryDashboard/TreasuryDashboard";
