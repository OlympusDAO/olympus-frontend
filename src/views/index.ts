export { default as Bond } from "./Bond/Bond";
export { default as ChooseBond } from "./ChooseBond/ChooseBond";
export { default as Stake } from "./Stake/Stake";
export { default as V1Stake } from "./V1-Stake/V1-Stake";
export { default as PoolTogether } from "./33Together/33together";
export { default as Wrap } from "./Wrap/Wrap";
export { default as TreasuryDashboard } from "./TreasuryDashboard/TreasuryDashboard";
export { default as Zap } from "./Zap/Zap";
