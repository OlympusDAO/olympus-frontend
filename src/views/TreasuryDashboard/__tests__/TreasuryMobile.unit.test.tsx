import { Environment } from "src/helpers/environment/Environment/Environment";
import { createMatchMedia } from "src/testHelpers";
import { render, screen } from "src/testUtils";

import TreasuryDashboard from "../TreasuryDashboard";

beforeAll(() => {
  window.matchMedia = createMatchMedia("300px");
});

describe("<TreasuryDashboard/> Mobile", () => {
  it("should render component", () => {
    const { container } = render(<TreasuryDashboard />);
    expect(container).toMatchSnapshot();
  });

  it("should render Metrics Dashboard only when Multifarm is disabled", () => {
    Environment.isMultifarmDashboardEnabled = jest.fn().mockReturnValue(undefined);
    expect(screen.queryByText("Revenue")).not.toBeInTheDocument(); //This is a string that is part of the tabs component
  });
});
