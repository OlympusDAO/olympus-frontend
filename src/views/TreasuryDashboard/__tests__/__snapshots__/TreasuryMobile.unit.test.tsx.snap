// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreasuryDashboard/> Mobile should render component 1`] = `
<div>
  <div
    class="smaller very-small"
    id="treasury-dashboard-view"
  >
    <div
      class="MuiTabs-root Tabs-root  css-u2rngf-MuiTabs-root"
    >
      <div
        class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
        style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
      />
      <div
        class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-1qz90fd-MuiTabs-scroller"
        style="margin-bottom: 0px;"
      >
        <div
          aria-label="dashboard-tabs"
          class="MuiTabs-flexContainer css-1pyy021-MuiTabs-flexContainer"
          role="tablist"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-ao3601-MuiTypography-root-MuiLink-root"
            href="#/dashboard"
            tabindex="0"
            textcolor="primary"
            value="0"
          >
            <button
              aria-controls="simple-tabpanel-undefined"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-rzx4ff-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-undefined"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Dashboard
            </button>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-ao3601-MuiTypography-root-MuiLink-root"
            href="#/dashboard/treasury"
            textcolor="primary"
            value="1"
          >
            <button
              aria-controls="simple-tabpanel-undefined"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-rzx4ff-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-undefined"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Treasury
            </button>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-ao3601-MuiTypography-root-MuiLink-root"
            href="#/dashboard/revenue"
            textcolor="primary"
            value="2"
          >
            <button
              aria-controls="simple-tabpanel-undefined"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-rzx4ff-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-undefined"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Revenue
            </button>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-ao3601-MuiTypography-root-MuiLink-root"
            href="#/dashboard/olympuspro"
            textcolor="primary"
            value="3"
          >
            <button
              aria-controls="simple-tabpanel-undefined"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-rzx4ff-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-undefined"
              role="tab"
              style="white-space: nowrap;"
              tabindex="-1"
              type="button"
            >
              Olympus Pro
            </button>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-ao3601-MuiTypography-root-MuiLink-root"
            href="#/dashboard/proteus"
            textcolor="primary"
            value="4"
          >
            <button
              aria-controls="simple-tabpanel-undefined"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-rzx4ff-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-undefined"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Proteus
            </button>
          </a>
        </div>
        <span
          class="MuiTabs-indicator css-gi9wfz-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-e057jq-MuiContainer-root"
      style="padding-left: 0px; padding-right: 0px;"
    >
      <div
        class="MuiBox-root css-tr0r3x"
      >
        <div
          class="hero-metrics MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card css-m19e0g-MuiPaper-root"
            style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-rjww1q"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1wlo379-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root  css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            Market Cap
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root  css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            OHM Price
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root wsoprice css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            gOHM Price
                            <div
                              class="MuiBox-root css-w341jy"
                              style="font-size: 14px;"
                            >
                              <div
                                class="MuiBox-root css-0"
                                style="display: inline-flex; justify-content: center; align-self: center;"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall info-icon css-1fx2bp7-MuiSvgIcon-root"
                                  focusable="false"
                                  style="margin: 0px 5px; font-size: 1em;"
                                  viewBox="0 0 20 20"
                                >
                                  <path
                                    d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root  css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            Circulating Supply (total)
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root  css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            Liquid Backing per OHM
                            <div
                              class="MuiBox-root css-w341jy"
                              style="font-size: 14px;"
                            >
                              <div
                                class="MuiBox-root css-0"
                                style="display: inline-flex; justify-content: center; align-self: center;"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall info-icon css-1fx2bp7-MuiSvgIcon-root"
                                  focusable="false"
                                  style="margin: 0px 5px; font-size: 1em;"
                                  viewBox="0 0 20 20"
                                >
                                  <path
                                    d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wmkgw3-MuiGrid-root"
                    >
                      <div
                        class="Metric-root  css-1xkrsym"
                      >
                        <div
                          class="MuiBox-root css-g45inh"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-label css-tlko1g-MuiTypography-root"
                          >
                            Current Index
                            <div
                              class="MuiBox-root css-w341jy"
                              style="font-size: 14px;"
                            >
                              <div
                                class="MuiBox-root css-0"
                                style="display: inline-flex; justify-content: center; align-self: center;"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall info-icon css-1fx2bp7-MuiSvgIcon-root"
                                  focusable="false"
                                  style="margin: 0px 5px; font-size: 1em;"
                                  viewBox="0 0 20 20"
                                >
                                  <path
                                    d="M 10 20 C 15.475 20 20 15.473 20 10 C 20 4.518 15.473 0 9.991 0 C 4.516 0 0 4.518 0 10 C 0 15.475 4.527 20 10 20 Z M 10 18.705 C 5.189 18.714 1.287 14.812 1.297 10.001 C 1.28 5.189 5.179 1.281 9.991 1.285 C 14.807 1.28 18.714 5.182 18.714 9.999 C 18.719 14.811 14.813 18.712 10 18.702 Z M 9.941 6.242 C 10.559 6.242 11.038 5.763 11.038 5.156 C 11.043 4.547 10.549 4.053 9.94 4.058 C 9.334 4.057 8.842 4.55 8.844 5.156 C 8.843 5.761 9.337 6.249 9.941 6.242 Z M 8.216 15.444 L 12.303 15.444 C 12.623 15.444 12.871 15.204 12.871 14.895 C 12.87 14.584 12.615 14.334 12.303 14.338 L 10.868 14.338 L 10.868 8.754 C 10.868 8.346 10.658 8.065 10.269 8.065 L 8.345 8.065 C 7.919 8.045 7.631 8.493 7.826 8.872 C 7.925 9.065 8.128 9.182 8.345 9.172 L 9.652 9.172 L 9.652 14.338 L 8.216 14.338 C 7.905 14.334 7.649 14.584 7.648 14.895 C 7.648 15.204 7.897 15.444 8.216 15.444 Z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 Metric-metric css-69sme-MuiTypography-root"
                            style="width: 100%;"
                          >
                            <span
                              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1l7q9tc-MuiSkeleton-root"
                              style="width: 100%;"
                            />
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 data-grid css-pcr0fy-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6 css-ebdhj4-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 Paper-root ohm-card ohm-chart-card css-m19e0g-MuiPaper-root"
              style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1y6dg9a-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-159wrwf-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                    style="display: flex; align-items: center; justify-content: center;"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-2r4182-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
